src/transactions/TransactionFrame.cpp

```diff
@@ -2120,14 +2122,18 @@ TransactionFrame::processRefundAndEmitFeeEvent(
     LedgerTxn ltx(ltxOuter);
     int64_t refund = refundSorobanFee(ltx, feeSource, txResult);
 
-    // TODO: handle fee event logic here
-
     meta.pushTxChangesAfter(ltx.getChanges());
     ltx.commit();
 
     // Emit fee refund event. A refund counts as a negative amount of fee
     // charged.
-    txEventManager.newFeeEvent(feeSource, -refund);
+    auto stage = TransactionEventStage::TRANSACTION_EVENT_STAGE_AFTER_TX;
+    if (protocolVersionStartsFrom(ltxOuter.loadHeader().current().ledgerVersion,
```

Comment:
I've just realized the CAP is likely mis-specified. I think only Soroban transactions will have TRANSACTION_EVENT_STAGE_AFTER_ALL_TXS stage in p23. I believe classic will remain as is.
@sisuresh to confirm.

Reply:
That's true, but you can only get to this line if this is a soroban tx.

Reply:
Ah, right, I guess it's getting a bit too late here, sorry.